### PR TITLE
Introduced ftJSONFormatter in order to use logrus with FT logging format

### PR DESCRIPTION
**Pull request by izzyblues**

### CODE REVIEW

File: formatter.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		case error:
+			// Otherwise errors are ignored by `encoding/json`
+			data[k] = v.Error()
+		default:
+			data[k] = v
+		}
+	}
+
+	if _, found := data[fieldKeyTime]; found {
+		data[fieldKeyLogTime] = entry.Time.Format(timestampFormat)
```
Comment (BerniVarga) [inline formatter.go]: I think we can remove this case from here. I would only leave the check for the @time field. If it's in the entries, let's just keep it, otherwise let's overwrite it with the actual formatted time. The logTime - or any other time related field can be added by the service that decides to do that.

File: logger.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+var formatter = newFTJSONFormatter()
+
+func init() {
+	log.Formatter = formatter
+}
+
+func InitLogger(serviceName string, logLevel string) {
+	parsedLogLevel, err := logrus.ParseLevel(logLevel)
+	if err != nil {
+		log.WithFields(logrus.Fields{"logLevel": logLevel, "err": err}).Fatal("Incorrect log level. Using INFO instead.")
```
Comment (peteclark-ft) [inline logger.go]: Will `Fatal` not exit the process?

File: README.md [hunk 3]
```diff
@@ -9,16 +9,22 @@ When working with this logger library, please make sure you use one of the init
 Note: You can still create your own standard logger by using the `NewLogger` function (check [Logrus](https://github.com/sirupsen/logrus/blob/master/logger.go#L69) for more details).
 
 
-### Creating an Entry
-The library maintains most of the Logrus's way of logging, but it adds some default fields (`@time`, `transaction_id` and `service_name`) for the created logging entries.
-For entry creation you can use the following methods:
-- `NewEntry` - with `transaction_id` as a parameter
-- `NewMonitoringEntry` - with `transaction_id`, `eventName` and `contentType` as parameters. Beside these, a `monitoring_event=true` field will also be added to the entry. This message will be picked up by the monitoring services and dashboards.
+### Logging a Monitoring Event
+The library maintains most of the Logrus's way of logging, but it adds some default fields
```
Comment (peteclark-ft) [inline README.md]: Minor grammar tweak - "The library is `logrus` compatible, but it includes a few default fields which help facilitate the monitoring of key application events."

File: README.md [hunk 4]
```diff
@@ -9,16 +9,22 @@ When working with this logger library, please make sure you use one of the init
...
-The library maintains most of the Logrus's way of logging, but it adds some default fields (`@time`, `transaction_id` and `service_name`) for the created logging entries.
-For entry creation you can use the following methods:
-- `NewEntry` - with `transaction_id` as a parameter
-- `NewMonitoringEntry` - with `transaction_id`, `eventName` and `contentType` as parameters. Beside these, a `monitoring_event=true` field will also be added to the entry. This message will be picked up by the monitoring services and dashboards.
+### Logging a Monitoring Event
+The library maintains most of the Logrus's way of logging, but it adds some default fields 
+(`@time`, `transaction_id` and `service_name`) to facilitate monitoring of relevant application events.
+You can add a monitoring event to a log entry by using the following method:
+- `WithMonitoringEvent` - with `transaction_id`, `eventName` and `contentType` as parameters. 
+Beside these, a `monitoring_event=true` field will also be added to the entry.
```
Comment (peteclark-ft) [inline README.md]: I'd keep this shorter and just say "A `monitoring_event=true` field will also be added to the entry."

File: README.md [hunk 5]
```diff
@@ -9,16 +9,22 @@ When working with this logger library, please make sure you use one of the init
...
+This message will be picked up by the monitoring services and dashboards.
 
 ### Adding additional fields to the Entry
 
-Beside the With... fields offered by the original Logrus Entry, the `WithUUID`, `WithTime` and `WithValidFlag` methods can also be used.
-The validation flag should mark whether the message is accepted as valid by the service it uses. Invalid messages will be ignored by some of the monitoring statistics (SLAs).
+Beside the With... fields offered by the original Logrus Entry, the following methods can be used:
+- `WithTransactionID`, to add a transaction ID to the log entry;
+- `WithUUID`, To add a UUID to the log entry;
+- `WithTime`, to set a custom time of the logging entry;
```
Comment (peteclark-ft) [inline README.md]: It might be worth mentioning that this will influence the splunk log time (if that is the case)

File: README.md [hunk 6]
```diff
@@ -9,16 +9,22 @@ When working with this logger library, please make sure you use one of the init
...
 
 ### Adding additional fields to the Entry
 
-Beside the With... fields offered by the original Logrus Entry, the `WithUUID`, `WithTime` and `WithValidFlag` methods can also be used.
-The validation flag should mark whether the message is accepted as valid by the service it uses. Invalid messages will be ignored by some of the monitoring statistics (SLAs).
+Beside the With... fields offered by the original Logrus Entry, the following methods can be used:
+- `WithTransactionID`, to add a transaction ID to the log entry;
+- `WithUUID`, To add a UUID to the log entry;
+- `WithTime`, to set a custom time of the logging entry; 
+- `WithValidFlag` to mark the if a message processed by an application is valid.
```
Comment (peteclark-ft) [inline README.md]: "to mark if a message processed by an application succeeded or not" ?

File: README.md [hunk 7]
```diff
@@ -10,20 +10,20 @@ Note: You can still create your own standard logger by using the `NewLogger` fun
...
 
 Beside the With... fields offered by the original Logrus Entry, the following methods can be used:
 - `WithTransactionID`, to add a transaction ID to the log entry;
-- `WithUUID`, To add a UUID to the log entry;
-- `WithTime`, to set a custom time of the logging entry; 
-- `WithValidFlag` to mark the if a message processed by an application is valid. 
+- `WithUUID`, to add a UUID to the log entry;
+- `WithTime`, to set a custom time of the logging entry (this can be used to influence Splunk log time); 
+- `WithValidFlag` to mark if a message processed by an application succeeded or not. 
 Invalid messages will be ignored by some of the monitoring statistics (SLAs).
```
[the remark this answers is not in the document]
Comment (izzyblues) [reply README.md]: Let me rephrase it properly according to the document made by Bernie
